app/src/main/java/com/aaronjwood/portauthority/network/Wireless.java

```diff
@@ -152,13 +154,47 @@ public String getInternalWifiIpAddress() {
      * @return Internal Wifi Subnet Netmask
      */
     public int getInternalWifiSubnet() {
-        // ToDo: Make sure this works with static DHCP reservations.
-        //FIXME this seems to have issues on some phones running 5.0+ https://code.google.com/p/android/issues/detail?id=82477
         WifiManager wifiManager = this.getWifiManager();
         DhcpInfo dhcpInfo = wifiManager.getDhcpInfo();
-        return dhcpInfo.netmask; //FIXME returns 0
+        int netmask = Integer.bitCount(dhcpInfo.netmask);
+        /*
+         * Workaround for #82477
+         * https://code.google.com/p/android/issues/detail?id=82477
+         * If dhcpInfo returns a subnet that cannot exist, then
+         * look up the Network interface instead.
+         */
+        if (dhcpInfo.netmask < 8 || dhcpInfo.netmask > 32) {
+            try {
+                InetAddress inetAddress = this.getWifiInetAddress();
+                NetworkInterface networkInterface = NetworkInterface.getByInetAddress(inetAddress);
+                for (InterfaceAddress address : networkInterface.getInterfaceAddresses()){
+                    if (inetAddress != null && inetAddress.equals(address.getAddress())){
+                        return address.getNetworkPrefixLength(); // This returns a short of the CIDR notation.
+                    }
+                }
+            } catch (SocketException e){
+                Log.d("SocketException", e.toString());
```

Comment:
Not sure if we wanna generally log in non-debug builds. It will just sit locally on someone's phone and only the people that know how to find it will be able to dig into their phone and look for it.